protocol/x/clob/keeper/process_single_match.go

```diff
@@ -432,18 +436,47 @@ func (k Keeper) persistMatchedOrders(
 	// keeper.
 
 	if err := k.subaccountsKeeper.TransferInsuranceFundPayments(ctx, insuranceFundDelta, perpetualId); err != nil {
-		return takerUpdateResult, makerUpdateResult, err
+		return takerUpdateResult, makerUpdateResult, affiliateRevSharesQuoteQuantums, err
+	}
+	perpetual, err := k.perpetualsKeeper.GetPerpetual(ctx, perpetualId)
+	if err != nil {
+		return takerUpdateResult, makerUpdateResult, affiliateRevSharesQuoteQuantums, err
+	}
+
+	fill := types.FillForProcess{
+		TakerAddr:             matchWithOrders.TakerOrder.GetSubaccountId().Owner,
+		TakerFeeQuoteQuantums: bigTakerFeeQuoteQuantums,
+		MakerAddr:             matchWithOrders.MakerOrder.GetSubaccountId().Owner,
+		MakerFeeQuoteQuantums: bigMakerFeeQuoteQuantums,
+		FillQuoteQuantums:     bigFillQuoteQuantums,
+		ProductId:             perpetualId,
+		MarketId:              perpetual.Params.MarketId,
+		MonthlyRollingTakerVolumeQuantums: k.statsKeeper.GetUserStats(
+			ctx,
+			matchWithOrders.TakerOrder.GetSubaccountId().Owner,
+		).TakerNotional,
 	}
 
 	// Distribute the fee amount from subacounts module to fee collector and rev share accounts
 	bigTotalFeeQuoteQuantums := new(big.Int).Add(bigTakerFeeQuoteQuantums, bigMakerFeeQuoteQuantums)
+	revshares, err := k.revshareKeeper.GetAllRevShares(ctx, fill)
+	affiliateRevSharesQuoteQuantums = big.NewInt(0)
+	for _, revshare := range revshares {
```

Comment:
Having to iterate this everytime isn't ideal, when we already have this info. Maybe update `GetAllRevShares` to return below object
```
type struct RevSharesForFill {
   affiliateRevShare types.RevShare,
   otherRevShares []types.RevShare,
}
```
Feel free to do what you think makes the most sense

Reply:
We need to iterate through this everytime anyway [here](https://github.com/dydxprotocol/v4-chain/blob/affan/revshare-logic/protocol/x/rewards/keeper/keeper.go#L140) and [here](https://github.com/dydxprotocol/v4-chain/blob/affan/revshare-logic/protocol/x/subaccounts/keeper/transfer.go#L251) - so this doesnt add too much overhead. I can think of a couple solutions - but feels either hacky or trying to overcomplicate it - can discuss offline further!

Reply:
Hmm, don't love the fact that we do all these iterations in `GetAllRevShare` just to do more iterations downstream. 
How about something like below: 
```
type struct RevSharesForFill {
  // All rev shares in a list
  AllRevShares []types.RevShare
  AffiliateRevShare types.RevShare
  FeeSourceToQuoteQuantums map[types.RevShareFeeSource]*big.Int
}
```
Note: we should never iterate through the maps, but only use them for accessing keys.